components/admin_console/schema_admin_settings.jsx

```diff
@@ -350,18 +350,50 @@ export default class SchemaAdminSettings extends React.Component {
 
     buildButtonSetting = (setting) => {
         const handleRequestAction = (success, error) => {
-            setting.action(success, error, this.state['ServiceSettings.SiteURL']);
+            const successCallback = (data) => {
+                const metadata = new Map();
+                Object.keys(data).forEach((key) => {
+                    metadata.set(key, data[key]);
+                });
+
+                const settings = (this.props.schema && this.props.schema.settings) || [];
```

Comment:
Does this instantiate and array ever from the code to the left of the `||` operator? Just asking because that looks like it would instantiate a boolean value.

Reply:
Yes, it checks if "this.props.schema" is defined and it is defined, if there is a valid array in "this.props.schema.settings", the it returns it. If either of them is undefined, then it returns the empty array. Same statement is used above: line  #108.

Reply:
Cool, I guess I'm just used to seeing this written like this so it confused me a bit:
```js
const settings = this.props.schema ? this.props.schema.settings : [];
```

Reply:
I assume the statement you mention (which i actually prefer, but used the already existing one since it achieved the same purpose in a similar context) is the way to go in most non-JS languages.

Reply:
If we had `'@babel/plugin-proposal-optional-chaining'` installed we could do:
```js
const settings = this.props.schema?.settings || [];
```